tests/components/zha/test_siren.py

```diff
@@ -0,0 +1,139 @@
+
+    # test that the state has changed from unavailable to off
+    assert hass.states.get(entity_id).state == STATE_OFF
+
+    # turn on from HA
+    with patch(
+        "zigpy.zcl.Cluster.request",
+        return_value=mock_coro([0x00, zcl_f.Status.SUCCESS]),
+    ):
+        # turn on via UI
+        await hass.services.async_call(
+            DOMAIN, "turn_on", {"entity_id": entity_id}, blocking=True
+        )
+        assert len(cluster.request.mock_calls) == 1
+        assert cluster.request.call_args[0][0] is False
+        assert cluster.request.call_args[0][1] == 0
+        assert cluster.request.call_args[0][3] == 54  # bitmask for default args
+        assert cluster.request.call_args[0][4] == 5  # duration in seconds
+        assert cluster.request.call_args[0][5] == 0
+        assert cluster.request.call_args[0][6] == 2
+
+    # test that the state has changed to on
+    assert hass.states.get(entity_id).state == STATE_ON
+
+    # turn off from HA
+    with patch(
+        "zigpy.zcl.Cluster.request",
+        return_value=mock_coro([0x01, zcl_f.Status.SUCCESS]),
+    ):
+        # turn off via UI
+        await hass.services.async_call(
+            DOMAIN, "turn_off", {"entity_id": entity_id}, blocking=True
+        )
+        assert len(cluster.request.mock_calls) == 1
+        assert cluster.request.call_args[0][0] is False
+        assert cluster.request.call_args[0][1] == 0
+        assert cluster.request.call_args[0][3] == 2  # bitmask for default args
+        assert cluster.request.call_args[0][4] == 5  # duration in seconds
+        assert cluster.request.call_args[0][5] == 0
+        assert cluster.request.call_args[0][6] == 2
+
+    # test that the state has changed to off
+    assert hass.states.get(entity_id).state == STATE_OFF
+
+    # turn on from HA
+    with patch(
+        "zigpy.zcl.Cluster.request",
+        return_value=mock_coro([0x00, zcl_f.Status.SUCCESS]),
+    ):
+        # turn on via UI
+        await hass.services.async_call(
+            DOMAIN,
+            "turn_on",
+            {
+                "entity_id": entity_id,
+                ATTR_DURATION: 10,
+                ATTR_TONE: WARNING_DEVICE_MODE_EMERGENCY_PANIC,
+                ATTR_VOLUME_LEVEL: WARNING_DEVICE_SOUND_MEDIUM,
+            },
+            blocking=True,
+        )
+        assert len(cluster.request.mock_calls) == 1
+        assert cluster.request.call_args[0][0] is False
+        assert cluster.request.call_args[0][1] == 0
+        assert cluster.request.call_args[0][3] == 101  # bitmask for passed args
+        assert cluster.request.call_args[0][4] == 10  # duration in seconds
+        assert cluster.request.call_args[0][5] == 0
+        assert cluster.request.call_args[0][6] == 2
+
+        # test that the state has changed to on
+    assert hass.states.get(entity_id).state == STATE_ON
+
+    now = dt_util.utcnow() + timedelta(seconds=15)
+    async_fire_time_changed(hass, now)
+    await hass.async_block_till_done()
```

Comment:
We should assert the state at the end of the test again.

Reply:
wow not sure how that got dropped... good catch